### PR TITLE
Support for Mixed Instances ASG in worker_groups_launch_template variable

### PR DESCRIPTION
**Pull request by sppwf**
# PR o'clock

## Description

Added support for Mixed Instances ASG in worker_groups_launch_template variable.
Changed the local.asg_tags - tag generation using for utility from terraform 12
worker_groups_launch_template_mixed variable left for backward compatibility

### Checklist

- [x] `terraform fmt` and `terraform validate` both work from the root and `examples/*` directories
- [x] CI tests are passing
- [x] I've added my change to CHANGELOG.md and highlighted any breaking changes
- [ ] README.md has been updated after any changes to variables and outputs. See https://github.com/terraform-aws-modules/terraform-aws-eks/#doc-generation

### CODE REVIEW

File: workers.tf [hunk 1]
```diff
@@ -325,16 +325,6 @@ resource "aws_iam_role_policy_attachment" "workers_additional_policies" {
   policy_arn = var.workers_additional_policies[count.index]
 }
 
-resource "null_resource" "tags_as_list_of_maps" {
```
Comment (max-rocket-internet) [inline workers.tf]: Nice!

File: examples/launch_templates/main.tf [hunk 2]
```diff
@@ -64,10 +64,19 @@ module "eks" {
       public_ip            = true
     },
     {
-      name                 = "worker-group-2"
+      name                 = "worker-spot-1"
```
Comment (max-rocket-internet) [inline examples/launch_templates/main.tf]: Can you leave this example as is but change the one in `examples/spot_instances` instead?

**Comment by max-rocket-internet** [comment]
If I understand correctly, we could remove anything related to `workers_launch_template_mixed`, right? That would be super cool 🙌

File: examples/launch_templates/main.tf (hunk 2 above)
Comment (sppwf) [reply examples/launch_templates/main.tf]: I will change it in two days as I am quite fur from good internet :)

**Comment by sppwf** [comment]
Yes, we can remove, however i just left it for backward compatibily.
Either we just create some migration plan, not sure how you guys used to do here. 

Thanks
Sergiu

**Comment by max-rocket-internet** [comment]
> Yes, we can remove, however i just left it for backward compatibily.

Great! Then please, if you can, do remove the launch template mixed related stuff. It was created as there was no other option in TF 0.11 but it would be much simpler and tidier if we just had `workers_launch_template` that covered all needs 🚀

File: examples/launch_templates/main.tf (hunk 2 above)
Comment (sppwf) [reply examples/launch_templates/main.tf]: I have made the changes

**Comment by sppwf** [comment]
@max-rocket-internet Should i do here something more, I am a little bit confuzed ?

**Comment by max-rocket-internet** [comment]
Just give me more time to have a look, it's quite a big change 😅

File: examples/spot_instances/main.tf [hunk 3]
```diff
@@ -56,15 +56,15 @@ module "eks" {
...
     {
-      name                    = "spot-1"
-      override_instance_types = ["m5.large", "m5a.large", "m5d.large", "m5ad.large"]
-      spot_instance_pools     = 4
-      asg_max_size            = 5
-      asg_desired_capacity    = 5
-      kubelet_extra_args      = "--node-labels=kubernetes.io/lifecycle=spot"
-      public_ip               = true
+      name                 = "spot-1"
+      instance_type        = "t2.medium"
```
Comment (max-rocket-internet) [inline examples/spot_instances/main.tf]: Can we leave a working example here with mixed spot instances? I would say having mixed instances in a spot ASG is very important so the spot requests definitely get fulfilled.

**Comment by max-rocket-internet** [comment]
I just tested this by running TF from `examples/spot_instances` but the ASG doesn't have any of the instance type overrides added:

Screen Shot 2019-09-03 at 15 15 09

**Comment by sppwf** [comment]
I was looking for this variable to be configured `on_demand_allocation_strategy`, however, specifing `override_instance_types` with some values means user will want spot instances.

**Comment by max-rocket-internet** [comment]
I tested again.

From `examples/spot_instances`:

```
module.eks.aws_autoscaling_group.workers_launch_template[0]: Creating...

Error: Error creating AutoScaling Group: ValidationError: Valid requests must contain either LaunchTemplate, LaunchConfigurationName, InstanceId or MixedInstancesPolicy parameter.
	status code: 400, request id: 33c4d651-d2f1-11e9-89e0-f7c70abdf77c

  on ../../workers_launch_template.tf line 3, in resource "aws_autoscaling_group" "workers_launch_template":
   3: resource "aws_autoscaling_group" "workers_launch_template" {
```

From `examples/launch_templates`:

```
module.eks.aws_autoscaling_group.workers_launch_template[1]: Creating...
module.eks.aws_autoscaling_group.workers_launch_template[0]: Creating...

Error: Error creating AutoScaling Group: ValidationError: Valid requests must contain either LaunchTemplate, LaunchConfigurationName, InstanceId or MixedInstancesPolicy parameter.
	status code: 400, request id: 35bb4f82-d2f1-11e9-ba6a-8da9e1bac398

  on ../../workers_launch_template.tf line 3, in resource "aws_autoscaling_group" "workers_launch_template":
   3: resource "aws_autoscaling_group" "workers_launch_template" {



Error: Error creating AutoScaling Group: ValidationError: Valid requests must contain either LaunchTemplate, LaunchConfigurationName, InstanceId or MixedInstancesPolicy parameter.
	status code: 400, request id: 35bb28fd-d2f1-11e9-ae7d-c78bc1196371

  on ../../workers_launch_template.tf line 3, in resource "aws_autoscaling_group" "workers_launch_template":
   3: resource "aws_autoscaling_group" "workers_launch_template" {
```

**Comment by max-rocket-internet** [comment]
Could you run `terraform apply` & `terraform destroy` from all 3 examples to ensure it works?

Really keen to get this PR merged though as it removes a lot of duplication 😄

**Comment by max-rocket-internet** [comment]
Thanks @sppwf !

**Pull request merged**